metadata/src/test/java/org/apache/kafka/metadata/RecordTestUtils.java

```diff
@@ -66,7 +66,18 @@ public static void replayAll(Object target,
                         Optional.class);
                     method.invoke(target, record, Optional.empty());
                 } catch (NoSuchMethodException t) {
-                    // ignore
+                    try {
+                        Method method = target.getClass().getMethod("replay",
+                            record.getClass(),
+                            long.class);
+                        method.invoke(target, record, 0L);
+                    } catch (NoSuchMethodException i) {
+                        // ignore
+                    } catch (InvocationTargetException i) {
```

Comment:
We have duplicated this pattern 3 times. I think we can remove this duplication if we add a `try` for the entire method. E.g. starts a line 57 and ends after the `for` loop.